controllers/propagator/propagation.go

```diff
@@ -901,6 +906,74 @@ func (r *PolicyReconciler) processTemplates(
 			ManagedClusterName: decision.ClusterName,
 		}
 
+		// Handle value encryption initialization
+		usesEncryption := templates.UsesEncryption(
+			policyT.ObjectDefinition.Raw, templateCfg.StartDelim, templateCfg.StopDelim,
+		)
+		// Initialize AES Key and initialization vector
+		if usesEncryption && !templateCfg.EncryptionEnabled {
+			log.V(1).Info("Found an object definition requiring encryption. Handling encryption keys.")
+			// Get/generate the encryption key
+			encryptionKey, err := r.getEncryptionKey(decision.ClusterName)
+			if err != nil {
+				log.Error(err, "Failed to get/generate the policy encryption key")
+
+				return err
+			}
+
+			// Get/generate the initialization vector
+			initializationVector, err := r.getInitializationVector(
+				rootPlc.GetName(), decision.ClusterName, annotations,
+			)
+			if err != nil {
+				log.Error(err, "Failed to get initialization vector")
+
+				return err
+			}
+
+			// Set the initialization vector in the annotations
+			replicatedPlc.SetAnnotations(annotations)
+
+			// Set the EncryptionConfig with the retrieved key
+			templateCfg.EncryptionConfig = templates.EncryptionConfig{
+				EncryptionEnabled:    true,
+				AESKey:               encryptionKey,
+				InitializationVector: initializationVector,
+			}
+
+			err = tmplResolver.SetEncryptionConfig(templateCfg.EncryptionConfig)
+			if err != nil {
+				log.Error(err, "Error setting encryption configuration")
+
+				return err
+			}
+		}
+
+		// Set initialization vector annotation on the ObjectDefinition for the controller's use
+		policyTObjectUnstructured := &unstructured.Unstructured{}
+
+		jsonErr := json.Unmarshal(policyT.ObjectDefinition.Raw, policyTObjectUnstructured)
+		if jsonErr != nil {
+			// it shouldn't get here but if it did just log a msg
+			// its all right, a generic msg will be used on the managedcluster
+			log.Error(jsonErr, "Error unmarshalling the object definition to JSON")
+		} else {
+			policyTAnnotations := policyTObjectUnstructured.GetAnnotations()
+			if policyTAnnotations == nil {
+				policyTAnnotations = make(map[string]string)
+			}
+			if policyIV, ok := annotations[ivAnnotation]; ok {
+				policyTAnnotations[ivAnnotation] = policyIV
+				policyTObjectUnstructured.SetAnnotations(policyTAnnotations)
+				updatedPolicyT, jsonErr := json.Marshal(policyTObjectUnstructured)
+				if jsonErr != nil {
+					log.Error(jsonErr, "Failed to marshall the policy template to JSON")
```

Comment:
I think you'll want to return the error here since the part that failed is what was once valid JSON that the IV annotation got added to. If it's no longer JSON serializable, then we don't want the rest of the code to execute since presumably, `ResolveTemplate` will succeed and the policy will be replicated only for the Config Policy Controller not to be able find the IV annotation.